Adding Dynatrace solution

keyoke opened the pull request:
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
Adding a new solution for Dynatrace

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)


-----------------------------------------------------------------------------------------------------------

keyoke commented:
@microsoft-github-policy-service agree company="Dynatrace"

v-spadarthi commented:
@keyoke : Could please resolve the below comments
1.Please provide the sample data(DynatraceAttacks_CL.json,DynatraceProblems_CL.json,DynatraceAuditLogs_CL.json,DynatraceSecurityProblems_CL.json) to test the Connector and workbook queries.
2.Replace text azure sentinel to Microsoft Sentinel wherever it is applicable.
All 4 data connectors having Azure sentinel please replace Microsoft Sentinel <img width="537" alt="image" src="https://user-images.githubusercontent.com/101796244/198224785-cc1aae3a-9c18-42ff-9a1a-79625ec3e90a.png">
3.Add workbookmetadata in below location 

4.In workbook after deploying we could see below, please provide sample data
<img width="487" alt="image" src="https://user-images.githubusercontent.com/101796244/198227719-a8bc1f19-47e4-4399-88f0-43f9abb07e7f.png">
Please use some preview images and charts in workbooks 
5.In input file please fix below 
<img width="728" alt="image" src="https://user-images.githubusercontent.com/101796244/198224206-24e32c1c-1722-4b45-a58d-f6f058c6183c.png">
Remove empty hunting queries property
Use version 2.0.0
I could see Is1Pconnector":true please make sure accordingly you can update the support details to Microsoft otherwise Is1Pconnector":False
<img width="536" alt="image" src="https://user-images.githubusercontent.com/101796244/198226897-01054fbd-de4a-4b39-acf6-b278f56d2eab.png">
6.Could you please let us know whether this solution will be published from your end or will it be published by us in our Partner Center.
If the solution is published from your Partner Center, the tier mentioned in solution metadata should be "Partner".
7.Package folder is missing and 2.0.0.zip as well
8.Logo is looking good validation passed 
<img width="830" alt="image" src="https://user-images.githubusercontent.com/101796244/198218383-9321ee23-9300-41c5-950a-08aea698c341.png">

For example If you need any information required, please see the below
ex : https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Dynamics%20365

keyoke commented:
Hi @v-spadarthi ,
Many Thanks for the feedback, I have committed the requested changes please let me know if anything else missing.
Regards
Gareth

v-spadarthi commented:
@keyoke : Sample data Looking fine
<img width="785" alt="image" src="https://user-images.githubusercontent.com/101796244/198602621-3d3bf64a-e9be-44f4-90fe-29814a5b26a5.png">
<img width="828" alt="image" src="https://user-images.githubusercontent.com/101796244/198615357-066f16cf-ce0c-45b1-b314-ac37ea3c0d39.png">
<img width="515" alt="image" src="https://user-images.githubusercontent.com/101796244/198605542-3722693e-e5e9-4a54-b48b-1b6a75742df9.png">
<img width="559" alt="image" src="https://user-images.githubusercontent.com/101796244/198606013-2591d7aa-fb90-4d38-853d-f5c5eef72939.png">
Parsers are Looking fine
<img width="785" alt="image" src="https://user-images.githubusercontent.com/101796244/198602867-39d34378-1c0d-4156-bb54-fb92aa762224.png">
<img width="955" alt="image" src="https://user-images.githubusercontent.com/101796244/198617160-e35413a4-232b-44a3-9b41-7479981408a4.png">
<img width="955" alt="image" src="https://user-images.githubusercontent.com/101796244/198612005-5e2b2922-4204-4a47-9314-34cf6a8e99c5.png">
<img width="960" alt="image" src="https://user-images.githubusercontent.com/101796244/198614946-183b8ad7-d05f-4196-81f3-9c5969d111da.png">

After testing the workbook, we could see below please fix the below errors
<img width="607" alt="image" src="https://user-images.githubusercontent.com/101796244/199002252-1dbb6d19-0918-4882-b19a-3e37b4983312.png">
<img width="484" alt="image" src="https://user-images.githubusercontent.com/101796244/199007775-3aede21b-ad9a-4081-a861-14c0055939bc.png">
Create UI Definition looking good
<img width="365" alt="image" src="https://user-images.githubusercontent.com/101796244/199009290-fc73d169-05f8-4ca2-837e-89bf8f341bd4.png">
Main template getting error please fix 
<img width="959" alt="image" src="https://user-images.githubusercontent.com/101796244/199010365-6ee477b2-c11a-4908-8d7b-d4d8fac19f1e.png">
Workbook details are empty in main template please see below
<img width="212" alt="image" src="https://user-images.githubusercontent.com/101796244/199012853-5d6dfdf6-b5c6-4ae0-8256-ccc2bfcad1bd.png">

keyoke commented:
@v-spadarthi - You right the workbook1-name parameter default value is not populated this seems to be related to this issue - #6141 , I have proposed the following change to resolve #6525

v-spadarthi commented:
@keyoke : Thanks for the fixes we will check and provide you an update.

v-spadarthi commented:
@keyoke : Thank you for the fixes.
In Dataconnector 
Connector_Dynatrace_Attacks.json deployed fine
<img width="767" alt="image" src="https://user-images.githubusercontent.com/101796244/200295784-6fdee85b-d7ba-494e-867f-68af5b208c84.png">
<img width="672" alt="image" src="https://user-images.githubusercontent.com/101796244/200309533-18401d13-2951-48ef-bf91-012cb9f28899.png">
Connector_Dynatrace_AuditLogs.json deployed fine
<img width="767" alt="image" src="https://user-images.githubusercontent.com/101796244/200294316-84b531ba-a39a-4d6d-9034-a0991426670d.png">
<img width="734" alt="image" src="https://user-images.githubusercontent.com/101796244/200310053-581603e4-3be9-4301-bb90-221cd7a6c0c3.png">
Connector_Dynatrace_Problems.json deployed fine
<img width="802" alt="image" src="https://user-images.githubusercontent.com/101796244/200294868-a1e616b3-44d2-468d-ab09-318361fb0b13.png">
<img width="721" alt="image" src="https://user-images.githubusercontent.com/101796244/200310277-3502c3a5-06ac-49be-9f3e-babb47465715.png">
Connector_Dynatrace_RuntimeVulnerabilities.json deployed fine
<img width="809" alt="image" src="https://user-images.githubusercontent.com/101796244/200295302-21fddc05-cd9c-46c2-aa89-0a16a296c55a.png">
<img width="772" alt="image" src="https://user-images.githubusercontent.com/101796244/200310755-d3a0c94e-dded-4be1-8340-1be3a833aacb.png">
While testing workbook still we could see below issues please fix it
Please update the sample data for this and provide us
<img width="500" alt="image" src="https://user-images.githubusercontent.com/101796244/200286304-34cd2ce2-73ff-4a67-b95f-4b63480630f4.png">
<img width="728" alt="image" src="https://user-images.githubusercontent.com/101796244/200286452-6a4eae76-d5c2-4739-8f43-7ea475cbd80f.png">
<img width="449" alt="image" src="https://user-images.githubusercontent.com/101796244/200286707-685f56ad-3864-45c1-b934-e06e9b2a8bc6.png">
<img width="540" alt="image" src="https://user-images.githubusercontent.com/101796244/200286855-f288a30a-4e43-413e-9ac2-ad8149bc8e6b.png">
Main template looking good
<img width="844" alt="image" src="https://user-images.githubusercontent.com/101796244/200288937-d07cde71-b775-454a-ae5d-4b1ca6f2cc57.png">
Arm-ttk also running fine
<img width="390" alt="image" src="https://user-images.githubusercontent.com/101796244/200289351-84f57a72-c3c1-40af-8fe7-b0549eb9dffc.png">

keyoke commented:
hi @v-spadarthi,
Thanks for the feedback, have committed some additional changes but can you confirm the function DynatraceAttacks looks like the following? One of the recent changes was with the parsers.

![image](https://user-images.githubusercontent.com/183547/200324855-2de82f19-2060-457a-9e07-32baa564a524.png)

Many Thanks

v-spadarthi commented:
@keyoke : Please resolve the below comments
  Sample data ingested fine
<img width="800" alt="image" src="https://user-images.githubusercontent.com/101796244/201039290-48eafdd3-9f36-4496-b28a-e3b69167489e.png">
Parser tested fine
<img width="803" alt="image" src="https://user-images.githubusercontent.com/101796244/201039822-a8925508-3359-4e0c-80d2-691a503d8d32.png">
In workbook we are getting below please fix it
<img width="547" alt="image" src="https://user-images.githubusercontent.com/101796244/201052985-480d42db-abb0-4e3a-9ee8-a12d90300b83.png">
<img width="666" alt="image" src="https://user-images.githubusercontent.com/101796244/201052766-e2cf73e6-7489-4975-8c86-92e37696d6a9.png">

v-spadarthi commented:
@keyoke : Please address the above workbook errors and conflicts as well. Thanks!
<img width="517" alt="image" src="https://user-images.githubusercontent.com/101796244/201826365-5d6c7e60-a081-4229-b4b5-3b8618d38895.png">

keyoke commented:
Hi @v-spadarthi, 
Sure I will deal with the conflicts, can you tell me in the screenshot you provide above if the schema is different for the older records in the table below?
Many Thanks

![image](https://user-images.githubusercontent.com/183547/202167733-5c785dcc-d88a-4a55-b0a9-e11e34fcf334.png)

v-spadarthi commented:
@keyoke : Please resolve the conflicts

v-spadarthi commented:
@keyoke : Please resolve the below workbook error still we are getting the same
<img width="957" alt="image" src="https://user-images.githubusercontent.com/101796244/203015390-81ba7319-8895-47ec-a00c-777d69d1eee1.png">

keyoke commented:
> @keyoke : Please resolve the below workbook error still we are getting the same <img alt="image" width="957" src="https://user-images.githubusercontent.com/101796244/203015390-81ba7319-8895-47ec-a00c-777d69d1eee1.png">

As far as I can see with latest updates this should be resolved according to my testing

v-spadarthi commented:
@keyoke : We are getting same issue please fix it
<img width="923" alt="image" src="https://user-images.githubusercontent.com/101796244/203903480-4f51b98e-1225-476f-b7d3-558c76efef3a.png">

v-spadarthi commented:
@keyoke : Please address the above comments.

keyoke commented:
> @keyoke : Please address the above comments.

@v-spadarthi  unfortunately IM not able to replicate the issue I followed the following steps

1 . Created new Sentinel and Log Analytics workspace
2.  Deploy the maintemplate.json generated by solution packaging to the new sentinel workspace
3.  Updated Azure-Sentinel\Tools\Sample-Data-Ingest-Tool\config.txt with workspace keys etc
4.  Copied Azure-Sentinel\Sample Data\Dynatrace*_CL.json to the Custom\ folder
5.  Run sample data import using Azure-Sentinel\Tools\Sample-Data-Ingest-Tool\
6. Wait and confirm custom tables and data exist
7.  View Dynatrace workbook

All steps above execute correctly the final result is

![image](https://user-images.githubusercontent.com/183547/205003634-1bd9027c-013f-4e83-8330-35f5846a34fd.png)

v-spadarthi commented:
@keyoke : Please fix the validation errors
<img width="955" alt="image" src="https://user-images.githubusercontent.com/101796244/205218024-327f9765-98dc-4ebd-bc85-fcd46889f464.png">

keyoke commented:
> @keyoke : Please fix the validation errors <img alt="image" width="955" src="https://user-images.githubusercontent.com/101796244/205218024-327f9765-98dc-4ebd-bc85-fcd46889f464.png">

@v-spadarthi  & @v-amolpatil I believe this is a breaking change when it comes to PR validation builds? the logo is still incoming into main branch and does not yet exist therefore the logo link will be invalid until merged. https://github.com/Azure/Azure-Sentinel/blob/master/.script/hyperLinkValidator.ts

v-spadarthi commented:
@keyoke : we have fixed now hyperlink validation issue.
Could you please fix the below workbook issue. Thanks

v-spadarthi commented:
@keyoke : Please address the above comments. Thanks!
@manishkumar1991 : Please review the Playbooks and provide your approval. Thanks!
@aprakash13 : Please review the Analytic Rules and provide your feedback. Thanks!

aprakash13 reviewed (commented):
Looks good from a query perspective. Please feel free to merge if everything else looks good.
For future , I would suggest that for detections that have just a single condition, such as | where VulnerabilityType != "CODE_LEVEL" that you move these all into a single custom alert that changes the alert name and description based on the content of the query -

manishkumar1991 commented:
@keyoke ,
Please explain, why this PR is having changes for crowd strike solution. 

![image](https://user-images.githubusercontent.com/97503740/207230502-8c00ab25-a2da-448a-baa1-de48c67367a7.png)

keyoke commented:
> @keyoke , Please explain, why this PR is having changes for crowd strike solution.
> 
> ![image](https://user-images.githubusercontent.com/97503740/207230502-8c00ab25-a2da-448a-baa1-de48c67367a7.png)

@manishkumar1991 - there should be no changes to any other solution at least none which I made knowingly likely a bad merge... they can be undone

manishkumar1991 commented:
@keyoke 
Remove crowd strike falcon from your PR and check below suggestion .

**************************************************************************
Once playbook is installed, changing secret id value is very difficult , as we need to get into each logic app and change manually .
So for this we recommend, use the key vault and fetch the secret key from there,  
But make sure , don't create new key vault resource using same playbook arm template, as we dont support key vault resource creation in our solution packaging right now .
so recommended way is to use the pre-existing key vault and draft the steps in readme files on how to save secret in key vault.

for example and proper illustration on how to use key vault , please check below playbook GitHub link

https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/ThreatXCloud/Playbooks/ThreatXPlaybooks/ThreatX-BlockIP-URL


Change all the playbook arm template files name to azuredeploy.json

In all Readme files , Kindly mention on how to get Access Token from dynatrace portal

keyoke commented:
> @keyoke Remove crowd strike falcon from your PR and check below suggestion .
> 
> Once playbook is installed, changing secret id value is very difficult , as we need to get into each logic app and change manually . So for this we recommend, use the key vault and fetch the secret key from there, But make sure , don't create new key vault resource using same playbook arm template, as we dont support key vault resource creation in our solution packaging right now . so recommended way is to use the pre-existing key vault and draft the steps in readme files on how to save secret in key vault.
> 
> for example and proper illustration on how to use key vault , please check below playbook GitHub link
> 
> https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/ThreatXCloud/Playbooks/ThreatXPlaybooks/ThreatX-BlockIP-URL
> 
> Change all the playbook arm template files name to azuredeploy.json
> 
> In all Readme files , Kindly mention on how to get Access Token from dynatrace portal

@manishkumar1991  I have made the requested changes cheers

v-mchatla commented:
Hi @keyoke,
If you have completed incorporating the changes suggested by @manishkumar1991. Please request for review.
Thanks

keyoke commented:
> @manishkumar1991 I have made the requested changes cheers



> Hi @keyoke, If you have completed incorporating the changes suggested by @manishkumar1991. Please request for review. Thanks

Hi @v-mchatla - yes I have completed the changes, @manishkumar1991  - can you review please? thanks

manishkumar1991 commented:
> > @manishkumar1991 I have made the requested changes cheers
> 
> > Hi @keyoke, If you have completed incorporating the changes suggested by @manishkumar1991. Please request for review. Thanks
> 
> Hi @v-mchatla - yes I have completed the changes, @manishkumar1991 - can you review please? thanks

I have started the review, allow some time

v-spadarthi commented:
@manishkumar1991 : Could you please review the PR and provide your comments. Thanks!

v-spadarthi commented:
@manishkumar1991 : Could you please review the PR and provide your comments. Thanks!

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Add_DynatraceApplicationSecurityAttackSourceIpThreatIntelligence/readme.md:
```diff
@@ -0,0 +1,46 @@
+## Initial Setup
+
+Logic App is utilitzed by Automation rules. You must setup an Automation rule. Go to the Automation Rules blade in Azure Sentinel. 
+
```
Please elaborate more , on how customer or user can attach this playbook with automation rule , alert or incident .
Give the steps by step

keyoke replied:
@manishkumar1991 - I have added additional steps

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Enrich-DynatraceAppSecAttackMSDefender365/readme.md:
```diff
@@ -0,0 +1,24 @@
+
+## Initial Setup
+
+Logic App is utilitzed by Automation rules. You must setup an Automation rule. Go to the Automation Rules blade in Azure Sentinel. 
```
Please elaborate more , on how customer or user can attach this playbook with automation rule , alert or incident .
Give the steps by step

keyoke replied:
@manishkumar1991 - I have added additional steps

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Enrich-DynatraceAppSecAttackWithSecurityAlerts/readme.md:
```diff
@@ -0,0 +1,24 @@
+## Initial Setup
+
+Logic App is utilitzed by Automation rules. You must setup an Automation rule. Go to the Automation Rules blade in Azure Sentinel. 
+
```
Please elaborate more , on how customer or user can attach this playbook with automation rule , alert or incident .
Give the steps by step

keyoke replied:
@manishkumar1991 - I have added additional steps

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Enrich_DynatraceApplicationSecurityAttackIncident/readme.md:
```diff
@@ -0,0 +1,22 @@
+## Initial Setup
+
+Logic App is utilitzed by Automation rules. You must setup an Automation rule. Go to the Automation Rules blade in Azure Sentinel. 
+
```
Please elaborate more , on how customer or user can attach this playbook with automation rule , alert or incident .
Give the steps by step

keyoke replied:
@manishkumar1991 - I have added additional steps

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Ingest-DynatraceMSDefender365/readme.md:
```diff
@@ -0,0 +1,22 @@
+
+## Initial Setup
+
+Logic App is utilitzed by Automation rules. You must setup an Automation rule. Go to the Automation Rules blade in Azure Sentinel. 
```
Please elaborate more , on how customer or user can attach this playbook with automation rule , alert or incident .
Give the steps by step

keyoke replied:
@manishkumar1991 - I have added additional steps

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Ingest-DynatraceSentinelSecurityAlerts/readme.md:
```diff
@@ -0,0 +1,22 @@
+## Initial Setup
+
+Logic App is utilitzed by Automation rules. You must setup an Automation rule. Go to the Automation Rules blade in Azure Sentinel. 
+
```
Please elaborate more , on how customer or user can attach this playbook with automation rule , alert or incident .
Give the steps by step

keyoke replied:
@manishkumar1991 - I have added additional steps

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Enrich-DynatraceAppSecAttackMSDefender365/azuredeploy.json:
```diff
@@ -0,0 +1,974 @@
+                            "azuremonitorlogs": {
+                                "connectionId": "[resourceId('Microsoft.Web/connections', variables('AzureMonitorLogsConnectionName'))]",
+                                "connectionName": "[variables('AzureMonitorLogsConnectionName')]",
+                                "id": "[subscriptionResourceId('Microsoft.Web/locations/managedApis', resourceGroup().location,'azuremonitorlogs')]"
```
Please check if azuremonitorlogs can also be a "ManagedServiceIdentity"

keyoke replied:
> azuremonitorlogs

Hi @manishkumar1991 ,
as far as I am aware its not supported? do you have a working example? I get the following error when trying to switch to managed identity

> {
    "status": "Failed",
    "error": {
        "code": "InvalidApiConnectionAlternativeParameters",
        "message": "The API connection 'azuremonitorlogs-EnrichDynatraceApplicationSecurityAttackIncident' has invalid inputs. Parameter value type cannot be set to 'Alternative' because managed identity is not supported in the API definition's connection parameters."
    }
}

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Enrich-DynatraceAppSecAttackWithSecurityAlerts/azuredeploy.json:
```diff
@@ -0,0 +1,974 @@
+                            "azuremonitorlogs": {
+                                "connectionId": "[resourceId('Microsoft.Web/connections', variables('AzureMonitorLogsConnectionName'))]",
+                                "connectionName": "[variables('AzureMonitorLogsConnectionName')]",
+                                "id": "[subscriptionResourceId('Microsoft.Web/locations/managedApis', resourceGroup().location,'azuremonitorlogs')]"
```
Please check if azuremonitorlogs can also be a "ManagedServiceIdentity"

keyoke replied:
> Please check if azuremonitorlogs can also be a "ManagedServiceIdentity"

Hi @manishkumar1991 ,
as far as I am aware its not supported? do you have a working example? I get the following error when trying to switch to managed identity

> {
    "status": "Failed",
    "error": {
        "code": "InvalidApiConnectionAlternativeParameters",
        "message": "The API connection 'azuremonitorlogs-EnrichDynatraceApplicationSecurityAttackIncident' has invalid inputs. Parameter value type cannot be set to 'Alternative' because managed identity is not supported in the API definition's connection parameters."
    }
}

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Enrich_DynatraceApplicationSecurityAttackIncident/azuredeploy.json:
```diff
@@ -0,0 +1,641 @@
+                            "azuremonitorlogs": {
+                                "connectionId": "[resourceId('Microsoft.Web/connections', variables('AzureMonitorLogsConnectionName'))]",
+                                "connectionName": "[variables('AzureMonitorLogsConnectionName')]",
+                                "id": "[subscriptionResourceId('Microsoft.Web/locations/managedApis', resourceGroup().location,'azuremonitorlogs')]"
```
Please check if azuremonitorlogs can also be a "ManagedServiceIdentity"

keyoke replied:
> Please check if azuremonitorlogs can also be a "ManagedServiceIdentity"

Hi @manishkumar1991 ,
as far as I am aware its not supported? do you have a working example? I get the following error when trying to switch to managed identity

> {
    "status": "Failed",
    "error": {
        "code": "InvalidApiConnectionAlternativeParameters",
        "message": "The API connection 'azuremonitorlogs-EnrichDynatraceApplicationSecurityAttackIncident' has invalid inputs. Parameter value type cannot be set to 'Alternative' because managed identity is not supported in the API definition's connection parameters."
    }
}

manishkumar1991 commented on Solutions/Dynatrace/Playbooks/Enrich_DynatraceApplicationSecurityAttackIncident/azuredeploy.json:
```diff
@@ -0,0 +1,641 @@
+                                "connectionId": "[resourceId('Microsoft.Web/connections', variables('AzureLogAnalyticsDataCollectorConnectionName'))]",
+                                "connectionName": "[variables('AzureLogAnalyticsDataCollectorConnectionName')]",
+                                "id": "[subscriptionResourceId('Microsoft.Web/locations/managedApis', resourceGroup().location,'azureloganalyticsdatacollector')]"
+                            },
```
Please check if azureloganalyticsdatacollector can also be a "ManagedServiceIdentity"

keyoke replied:
> Please check if azureloganalyticsdatacollector can also be a "ManagedServiceIdentity"

Hi @manishkumar1991 ,
as far as I am aware its not supported? do you have a working example? I get the following error when trying to switch to managed identity

> {
    "status": "Failed",
    "error": {
        "code": "InvalidApiConnectionAlternativeParameters",
        "message": "The API connection 'azureloganalyticsdatacollector-EnrichDynatraceApplicationSecurityAttackIncident' has invalid inputs. Parameter value type cannot be set to 'Alternative' because managed identity is not supported in the API definition's connection parameters."
    }
}

manishkumar1991 commented:
I tried to give token the attacks.read permission , but there is no such permission in dynatrace , can you please double check on this , find the screenshot below :  

![image](https://user-images.githubusercontent.com/97503740/209315157-30a5101e-3fab-4666-8676-4d1d02c6515c.png)

I am creating token from going to "Manage" and then choosing "access tokens"  and generating one from there , 
Let me know if there is any other procedure to achieve the same 


Please check my all other comments as well,

v-spadarthi commented:
@keyoke : Please address the @manishkumar1991 comments. Thanks

keyoke commented:
> I tried to give token the attacks.read permission , but there is no such permission in dynatrace , can you please double check on this , find the screenshot below :
> 
> ![image](https://user-images.githubusercontent.com/97503740/209315157-30a5101e-3fab-4666-8676-4d1d02c6515c.png)
> 
> I am creating token from going to "Manage" and then choosing "access tokens" and generating one from there , Let me know if there is any other procedure to achieve the same
> 
> Please check my all other comments as well,

Hi @manishkumar1991 ,
Thanks for the feedback I will check your comments and make required changes. regarding the attacks permission, its only available for use if you have Application Security monitoring licensed for your tenant. A trial tenant does not have this capability out of the box. We have also updated to connector copy to reflect this.
Regards
Gareth

keyoke commented:
> @keyoke : Please address the @manishkumar1991 comments. Thanks

hi @v-spadarthi  ok have done so

aprakash13 reviewed (changes requested):
Couple of quick suggestions from an analytics perspective:

- Please have the description section contained within a single quote (' ').

- In the dataTypes field the value is : TableName_CL. While in the query the table name is used without _CL. Can you please check and verify if the query should also use TableName_CL. Generally if the query uses a parser the table name used in the query is same as the parser name used.

v-spadarthi commented:
@keyoke : Please address the @aprakash13 comments. Thanks!

keyoke commented:
> Couple of quick suggestions from an analytics perspective:
> 
> * Please have the description section contained within a single quote (' ').
> * In the dataTypes field the value is : TableName_CL. While in the query the table name is used without _CL. Can you please check and verify if the query should also use TableName_CL. Generally if the query uses a parser the table name used in the query is same as the parser name used.

Thanks I have quoted the descriptions and as we are using parsers I have corrected the data type names in the analytics rules

keyoke commented:
> @keyoke : Please address the @aprakash13 comments. Thanks!

done

manishkumar1991 commented:
@keyoke, Thanks for taking care of all the changes. 

As I don't have the access to live environment, so requesting you to please share the screenshot of working playbooks that shows all the actions present in playbook logic apps are working fine.

keyoke commented:
> Couple of quick suggestions from an analytics perspective:
> 
> * Please have the description section contained within a single quote (' ').
> * In the dataTypes field the value is : TableName_CL. While in the query the table name is used without _CL. Can you please check and verify if the query should also use TableName_CL. Generally if the query uses a parser the table name used in the query is same as the parser name used.

@aprakash13 is it expected that incidentConfiguration properties are ignored when creating a new analytics rule from a template?

![image](https://user-images.githubusercontent.com/183547/210513300-89d7c8b2-caf0-47c2-940d-974da73ae94e.png)

results in

![image](https://user-images.githubusercontent.com/183547/210513770-3d898363-aec4-4e5b-b8e1-798456b828c1.png)

keyoke commented:
> @keyoke, Thanks for taking care of all the changes.
> 
> As I don't have the access to live environment, so requesting you to please share the screenshot of working playbooks that shows all the actions present in playbook logic apps are working fine.

@manishkumar1991  - please see below screenshots

**Add_DynatraceApplicationSecurityAttackSourceIpThreatIntelligence**

<img width="1119" alt="image" src="https://user-images.githubusercontent.com/183547/210518591-46e25a95-47fe-470a-96a0-c0d0353f5d09.png">

**Enrich_DynatraceApplicationSecurityAttackIncident**

<img width="1112" alt="image" src="https://user-images.githubusercontent.com/183547/210520367-9655e7c1-3ae5-4d97-93ad-6b7c46db2982.png">

**Ingest-DynatraceMSDefender365**

<img width="1175" alt="image" src="https://user-images.githubusercontent.com/183547/210524726-b3dd4dc1-0b5a-4592-8abc-e3ba28efc7ba.png">

**Ingest-DynatraceSentinelSecurityAlerts**

<img width="1174" alt="image" src="https://user-images.githubusercontent.com/183547/210526084-cbb09532-3c3b-4abf-91df-d5c63104a18e.png">

**Enrich-DynatraceAppSecAttackMSDefender365**

<img width="1061" alt="image" src="https://user-images.githubusercontent.com/183547/210529295-4a7c0475-4c38-4967-ab88-0b7954f83877.png">

**Enrich-DynatraceAppSecAttackWithSecurityAlerts**

<img width="1079" alt="image" src="https://user-images.githubusercontent.com/183547/210531653-5a13caf5-b124-41c2-a47f-cd9af81d08e3.png">

manishkumar1991 reviewed (approved):
approved

v-atulyadav commented:
Looking into this

v-atulyadav reviewed (commented):
Hi @keyoke, in SolutionMetadata.json & MainTemplate.json, we found a special character in domain category. Please remove this special character and update the files. Thanks

<img width="335" alt="image" src="https://user-images.githubusercontent.com/104008048/212024939-13f081eb-6c00-4de6-9f39-245924105434.png">

keyoke commented:
> Hi @keyoke, in SolutionMetadata.json & MainTemplate.json, we found a special character in domain category. Please remove this special character and update the files. Thanks
> 
> <img alt="image" width="335" src="https://user-images.githubusercontent.com/104008048/212024939-13f081eb-6c00-4de6-9f39-245924105434.png">

Hi @v-atulyadav, ok thanks I have removed it!

keyoke commented:
@v-atulyadav validation builds have suddenly started faiIling due to what I understand is to be a false positive I have also created a new issue #7090

v-sabiraj commented:
@keyoke, please update the branch with master, this is now fixed and updated in master, thanks.

keyoke commented:
> @keyoke, please update the branch with master, this is now fixed and updated in master, thanks.

hi @v-sabiraj  - I have done so and pushed changes to my branch but seems PR validation still fails...

keyoke commented:
> @keyoke, please update the branch with master, this is now fixed and updated in master, thanks.

hi @v-sabiraj the pr validation issue is now resolved, thanks

v-atulyadav commented:
Hi @keyoke, thanks for the fix will check and provide you feedback. Thanks.

v-atulyadav commented:
Hi @keyoke, can you please remove below mentioned zip file from package rest looks good now. Thanks.

<img width="249" alt="image" src="https://user-images.githubusercontent.com/104008048/213689872-6b3e6aa6-0441-4bf5-8d1a-22f7cf41dc71.png">

keyoke commented:
> Hi @keyoke, can you please remove below mentioned zip file from package rest looks good now. Thanks.
> 
> <img alt="image" width="249" src="https://user-images.githubusercontent.com/104008048/213689872-6b3e6aa6-0441-4bf5-8d1a-22f7cf41dc71.png">

Hi @v-atulyadav  thanks I have deleted this file
